test/e2e/storage/pd.go

```diff
@@ -428,6 +428,8 @@ var _ = framework.KubeDescribe("Pod Disks", func() {
 			By("Cleaning up PD-RW test env")
 			podClient.Delete(host0Pod.Name, metav1.NewDeleteOptions(0))
 			detachAndDeletePDs(diskName, []types.NodeName{host0Name})
+			framework.WaitForNodeToBeReady(f.ClientSet, string(host0Name), nodeStatusTimeout)
+			Expect(len(nodes.Items)).To(Equal(initialGroupSize))
```

Comment:
Do you need to get refresh `nodes` again? Otherwise, this will pick up the old `nodes` array?